### PR TITLE
Add some logs for kubeadm join

### PR DESCRIPTION
**Pull request by ping035627**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
Add some logs for kubeadm join. It make us view the whole join process clearly.

Which issue(s) this PR fixes:

Fixes #

Special notes for your reviewer:

Does this PR introduce a user-facing change?:

/release-note-none
/priority backlog

### CODE REVIEW

**Review by yagonobre — COMMENTED** [review]
Hi @ping035627, thanks for your pr.
We should keep `klog.V(1)` or `fmt.Println` not both.

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 1]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
 	}
 	bootstrapKubeConfigFile := kubeadmconstants.GetBootstrapKubeletKubeConfigPath()
 
-	// Write the bootstrap kubelet config file or the TLS-Boostrapped kubelet config file down to disk
-	klog.V(1).Infoln("[kubelet-start] writing bootstrap kubelet config file at", bootstrapKubeConfigFile)
+	// Write the bootstrap kubelet config file or the TLS-Bootstrapped kubelet config file down to disk
+	fmt.Println("[kubelet-start] writing bootstrap kubelet config file at", bootstrapKubeConfigFile)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: use `fmt.Printf`
[1]

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 2]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
...
 	if err := kubeconfigutil.WriteToDisk(bootstrapKubeConfigFile, tlsBootstrapCfg); err != nil {
 		return errors.Wrap(err, "couldn't save bootstrap-kubelet.conf to disk")
 	}
 
 	// Write the ca certificate to disk so kubelet can use it for authentication
 	cluster := tlsBootstrapCfg.Contexts[tlsBootstrapCfg.CurrentContext].Cluster
 	if _, err := os.Stat(cfg.CACertPath); os.IsNotExist(err) {
+		fmt.Println("[kubelet-start] writing ca certificate at", cfg.CACertPath)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: [1]

File: cmd/kubeadm/app/cmd/phases/join/preflight.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func runPreflight(c workflow.RunData) error {
 			return errors.New(msg.String())
 		}
 
-		// run kubeadm init preflight checks for checking all the prequisites
+		// run kubeadm init preflight checks for checking all the requisites
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/preflight.go]: `prerequisites`

**Comment by neolit123** [comment]
@ping035627 unfortunately we've entered code freeze and now we cannot merge this PR until the 19th of march because it's not a critical-bug fix.

this is part of the standard k8s release cycle.

https://github.com/kubernetes/sig-release/tree/master/releases/release-1.14#timeline

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 4]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
 	}
 	bootstrapKubeConfigFile := kubeadmconstants.GetBootstrapKubeletKubeConfigPath()
 
-	// Write the bootstrap kubelet config file or the TLS-Boostrapped kubelet config file down to disk
-	klog.V(1).Infoln("[kubelet-start] writing bootstrap kubelet config file at", bootstrapKubeConfigFile)
+	// Write the bootstrap kubelet config file or the TLS-Bootstrapped kubelet config file down to disk
+	fmt.Printf("[kubelet-start] writing bootstrap kubelet config file at", bootstrapKubeConfigFile)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: this needs to be `fmt.Printf(........ at: %s, bootstrapKubeConfigFile)`

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 5]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
...
 	if err := kubeconfigutil.WriteToDisk(bootstrapKubeConfigFile, tlsBootstrapCfg); err != nil {
 		return errors.Wrap(err, "couldn't save bootstrap-kubelet.conf to disk")
 	}
 
 	// Write the ca certificate to disk so kubelet can use it for authentication
 	cluster := tlsBootstrapCfg.Contexts[tlsBootstrapCfg.CurrentContext].Cluster
 	if _, err := os.Stat(cfg.CACertPath); os.IsNotExist(err) {
+		fmt.Printf("[kubelet-start] writing ca certificate at", cfg.CACertPath)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: also missing %s

**Comment by neolit123** [comment]
code freeze is lifted @ping035627 
please see my latest comments and squash your commits into one.

**Comment by ping035627** [comment]
@neolit123 ok, I have corrected. 
sorry, I make such a low-level mistake. :(

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 6]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
 	}
 	bootstrapKubeConfigFile := kubeadmconstants.GetBootstrapKubeletKubeConfigPath()
 
-	// Write the bootstrap kubelet config file or the TLS-Boostrapped kubelet config file down to disk
-	klog.V(1).Infoln("[kubelet-start] writing bootstrap kubelet config file at", bootstrapKubeConfigFile)
+	// Write the bootstrap kubelet config file or the TLS-Bootstrapped kubelet config file down to disk
+	fmt.Printf("[kubelet-start] writing bootstrap kubelet config file at %s", bootstrapKubeConfigFile)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: i think we should still use V(1) for now.
please change to:
```
klog.V(1).Infof("[kubelet-start] writing bootstrap kubelet config file at %s", bootstrapKubeConfigFile)`
```

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 7]
```diff
@@ -101,15 +101,16 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
...
 	if err := kubeconfigutil.WriteToDisk(bootstrapKubeConfigFile, tlsBootstrapCfg); err != nil {
 		return errors.Wrap(err, "couldn't save bootstrap-kubelet.conf to disk")
 	}
 
 	// Write the ca certificate to disk so kubelet can use it for authentication
 	cluster := tlsBootstrapCfg.Contexts[tlsBootstrapCfg.CurrentContext].Cluster
 	if _, err := os.Stat(cfg.CACertPath); os.IsNotExist(err) {
+		fmt.Printf("[kubelet-start] writing ca certificate at %s", cfg.CACertPath)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]:
```
klog.V(1).Infof("[kubelet-start] writing CA certificate at %s", cfg.CACertPath)
```

File: cmd/kubeadm/app/cmd/phases/join/kubelet.go [hunk 8]
```diff
@@ -127,7 +128,7 @@ func runKubeletStartJoinPhase(c workflow.RunData) error {
 
 	// Configure the kubelet. In this short timeframe, kubeadm is trying to stop/restart the kubelet
 	// Try to stop the kubelet service so no race conditions occur when configuring it
-	klog.V(1).Infoln("[kubelet-start] Stopping the kubelet")
+	fmt.Println("[kubelet-start] Stopping the kubelet")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/kubelet.go]: i think we should keep the original, see the "start" step bellow.

File: cmd/kubeadm/app/cmd/phases/join/preflight.go [hunk 9]
```diff
@@ -87,7 +86,7 @@ func runPreflight(c workflow.RunData) error {
 	fmt.Println("[preflight] Running pre-flight checks")
 
 	// Start with general checks
-	klog.V(1).Infoln("[preflight] Running general checks")
+	fmt.Println("[preflight] Running general checks")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/preflight.go]: we should keep the original V(1) here adds more detail to the line `fmt.Println("[preflight] Running pre-flight checks")
` line above.
[1]

File: cmd/kubeadm/app/cmd/phases/join/preflight.go [hunk 10]
```diff
@@ -98,7 +97,7 @@ func runPreflight(c workflow.RunData) error {
 	}
 
 	// Continue with more specific checks based on the init configuration
-	klog.V(1).Infoln("[preflight] Running configuration dependant checks")
+	fmt.Println("[preflight] Running configuration dependant checks")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/join/preflight.go]: [1]

**Review by neolit123 — COMMENTED** [review]
@ping035627 no, problem.
i did some more evaluation here and added comments again.

the idea of the regular kubeadm execution output (without `-v=x`) is that we should print minimal information, using `fmt.Print*`. from there we allow the user to print more detail if they want using `-v=x`.

@kubernetes/sig-cluster-lifecycle-pr-reviews

**Review by yagonobre — COMMENTED** [review]
Thanks again @ping035627 
/ok-to-test
/lgtm

**Review by neolit123 — COMMENTED** [review]
thanks
/approve

**Pull request merged**